### ISSUE TITLE
Add HDF5 Native Output Support

### ISSUE THREAD

**Issue by wongooz** [ORNL-MDF/3DThesis#16]
There is interest in directly wrapping the outputs, especially the time-temperature histories, into an HDF5 container. Direct visualization in Paraview requires creating an accompanying XDMF file which is also beneficial.

**Comment by gknapp1** [comment]
@JamieStumpORNL implemented a draft of writing the output of the time-temperature histories to a single CSV in 46988b1. It's not quite HDF5 support, but does that help with your data processing pipeline?

**Comment by wongooz** [comment]
I have had issues compiling this fix that I'm working through, but conceptually it should make the process easier.

**Comment by JamieStumpORNL** [comment]
That branch ("JamieChanges") has mandatory mpi, so maybe just wait until it gets merged with the main branch where mpi isn't required (should compile the same as what you had before)

**Comment by gknapp1** [comment]
I'll work with Jamie to get the single output file feature pulled out of his branch and into main. That pull request will get linked to this issue, so you should get a notification once it is merged into main.

**Comment by gknapp1** [comment]
@wongooz: #20 was just merged into `master` and adds some of the functionality that you wanted if you run snapshot mode with `T_hist` output and no melt pool tracking. The output file is still CSV, but now contains columns for `x`, `y`, and `z` location plus one column per time step that you output temperatures (previously, each xyz location was output as a separate file). For example, if you set the simulation to output at three time steps (0.1, 0.3, and 0.5 s), then the output will look like:

```csv
x,y,z,0.1,0.3,0.5
0,0,0,1200,1300,800
0,1,0,1100,1200,700
0,2,0,1000,1050,650
...
```

**Comment by wongooz** [comment]
Sounds good. I will pull, compile, and check it out. Thank you!
